api/RestfulService.php

```diff
@@ -175,7 +191,7 @@ public function request($subURL = '', $method = "GET", $data = null, $headers =
 			$method,
 			$data,
 			array_merge((array)$this->customHeaders, (array)$headers),
-			$curlOptions + (array)$this->config()->default_curl_options,
+			$curlOptions + (array)$this->get_default_curl_options(),
```

Comment:
You're calling a static function dynamically here; don't you mean `static::get_default_curl_options()`?